diff --git configure configure
index bb13697..c26ea81 100755
--- configure
+++ configure
@@ -318,8 +318,7 @@ def handleNativeConfig(pargs, cc=None):
     # Check tools are present
     tools = { 'LINKER':'ld',
               'ARCHIVER':'ar',
-              'NM':'nm',
-              'OBJDUMP':'objdump'
+              'NM':'nm'
             }
     for (name, executable) in tools.items():
         if not getAbsPathForToolInPathEnv(executable):
@@ -378,8 +377,7 @@ def handleLLVMConfig(pargs, cc=None):
     llvmDeps = {
         'NM' : ['llvm-nm'],
         'ARCHIVER' : ['llvm-ar'],
-        'LINKER': ['llvm-link', 'llvm-ld' ], # Two possible tools to use as linker
-        'OBJDUMP': ['llvm-objdump']
+        'LINKER': ['llvm-link', 'llvm-ld' ] # Two possible tools to use as linker
     }
 
     for (name,toolNames) in llvmDeps.items():
diff --git libc/stdlib/stdlib.c libc/stdlib/stdlib.c
index ccd3c01..eead241 100644
--- libc/stdlib/stdlib.c
+++ libc/stdlib/stdlib.c
@@ -343,10 +343,10 @@ strong_alias(strtol,strtoimax)
 #else
 #undef strtoll
 #endif
-extern __typeof(strtol) __XL_NPP(strtoll);
-libc_hidden_proto(__XL_NPP(strtoll))
-strong_alias(__XL_NPP(strtol),__XL_NPP(strtoll))
-libc_hidden_def(__XL_NPP(strtoll))
+//extern __typeof(strtol) __XL_NPP(strtoll);
+//libc_hidden_proto(__XL_NPP(strtoll))
+//strong_alias(__XL_NPP(strtol),__XL_NPP(strtoll))
+//libc_hidden_def(__XL_NPP(strtoll))
 #endif
 
 #endif
@@ -397,10 +397,10 @@ strong_alias(strtoul,strtoumax)
 #else
 #undef strtoull
 #endif
-extern __typeof(strtoul) __XL_NPP(strtoull);
-libc_hidden_proto(__XL_NPP(strtoull))
-strong_alias(__XL_NPP(strtoul),__XL_NPP(strtoull))
-libc_hidden_def(__XL_NPP(strtoull))
+//extern __typeof(strtoul) __XL_NPP(strtoull);
+//libc_hidden_proto(__XL_NPP(strtoull))
+//strong_alias(__XL_NPP(strtoul),__XL_NPP(strtoull))
+//libc_hidden_def(__XL_NPP(strtoull))
 #endif
 
 
@@ -1060,10 +1060,10 @@ strong_alias(wcstol,wcstoimax)
 #else
 #undef wcstoll
 #endif
-extern __typeof(wcstol) __XL_NPP(wcstoll);
-libc_hidden_proto(__XL_NPP(wcstoll))
-strong_alias(__XL_NPP(wcstol),__XL_NPP(wcstoll))
-libc_hidden_def(__XL_NPP(wcstoll))
+//extern __typeof(wcstol) __XL_NPP(wcstoll);
+//libc_hidden_proto(__XL_NPP(wcstoll))
+//strong_alias(__XL_NPP(wcstol),__XL_NPP(wcstoll))
+//libc_hidden_def(__XL_NPP(wcstoll))
 #endif
 
 #endif
@@ -1114,10 +1114,10 @@ strong_alias(wcstoul,wcstoumax)
 #else
 #undef wcstoull
 #endif
-extern __typeof(wcstoul) __XL_NPP(wcstoull);
-libc_hidden_proto(__XL_NPP(wcstoull))
-strong_alias(__XL_NPP(wcstoul),__XL_NPP(wcstoull))
-libc_hidden_def(__XL_NPP(wcstoull))
+//extern __typeof(wcstoul) __XL_NPP(wcstoull);
+//libc_hidden_proto(__XL_NPP(wcstoull))
+//strong_alias(__XL_NPP(wcstoul),__XL_NPP(wcstoull))
+//libc_hidden_def(__XL_NPP(wcstoull))
 #endif
 
 #endif
