diff -rupN llvm-2.9/lib/Support/DynamicLibrary.cpp llvm-2.9-new/lib/Support/DynamicLibrary.cpp
--- llvm-2.9/lib/Support/DynamicLibrary.cpp	2010-12-07 02:56:20.000000000 -0500
+++ llvm-2.9-new/lib/Support/DynamicLibrary.cpp	2014-05-21 13:17:05.963537848 -0400
@@ -106,6 +106,9 @@ void *SearchForAddressOfSpecialSymbol(co
 }
 
 void* DynamicLibrary::SearchForAddressOfSymbol(const char* symbolName) {
+  // If the first character of the symbolName is '\x1' (asm sentinel), skip it.
+  if (symbolName[0] == '\x1')
+    symbolName = &symbolName[1];
   // First check symbols added via AddSymbol().
   if (ExplicitSymbols) {
     std::map<std::string, void *>::iterator I =
