diff -rupN stp/src/c_interface/c_interface.cpp stp.new/src/c_interface/c_interface.cpp
--- stp/src/c_interface/c_interface.cpp	2012-11-07 12:05:44.000000000 -0500
+++ stp.new/src/c_interface/c_interface.cpp	2015-01-27 16:26:24.380088210 -0500
@@ -25,7 +25,7 @@ typedef BEEV::BVSolver*
 typedef BEEV::AbsRefine_CounterExample * ctrexamplestar;
 typedef BEEV::ASTVec                     nodelist;
 typedef BEEV::CompleteCounterExample*    CompleteCEStar;
-BEEV::ASTVec *decls = NULL;
+__thread BEEV::ASTVec *decls = NULL;
 //vector<BEEV::ASTNode *> created_exprs;
 
 // persist holds a copy of ASTNodes so that the reference count of
diff -rupN stp/src/extlib-abc/aig/cnf/cnfCore.c stp.new/src/extlib-abc/aig/cnf/cnfCore.c
--- stp/src/extlib-abc/aig/cnf/cnfCore.c	2012-11-07 12:05:57.000000000 -0500
+++ stp.new/src/extlib-abc/aig/cnf/cnfCore.c	2015-01-27 17:08:24.778899183 -0500
@@ -24,7 +24,7 @@
 ///                        DECLARATIONS                              ///
 ////////////////////////////////////////////////////////////////////////

-static Cnf_Man_t * s_pManCnf = NULL;
+static __thread Cnf_Man_t * s_pManCnf = NULL;

 ////////////////////////////////////////////////////////////////////////
 ///                     FUNCTION DEFINITIONS                         ///
diff -rupN stp/src/main/Globals.cpp stp.new/src/main/Globals.cpp
--- stp/src/main/Globals.cpp	2012-11-07 12:05:47.000000000 -0500
+++ stp.new/src/main/Globals.cpp	2015-01-27 16:26:24.380088210 -0500
@@ -16,11 +16,11 @@ namespace BEEV
   enum inputStatus input_status = NOT_DECLARED;
 
   //Originally just used by the parser, now used elesewhere.
-  STP     * GlobalSTP;
-  STPMgr  * ParserBM;
+  __thread STP     * GlobalSTP;
+  __thread STPMgr  * ParserBM;
 
   // Used exclusively for parsing.
-  ParserInterface * parserInterface;
+  __thread ParserInterface * parserInterface;
 
   void (*vc_error_hdlr)(const char* err_msg) = NULL;
 
diff -rupN stp/src/main/Globals.h stp.new/src/main/Globals.h
--- stp/src/main/Globals.h	2012-11-07 12:05:47.000000000 -0500
+++ stp.new/src/main/Globals.h	2015-01-27 16:26:24.380088210 -0500
@@ -72,9 +72,9 @@ namespace BEEV
   extern ASTNode ASTFalse, ASTTrue, ASTUndefined;
 
   //Useful global variables. Use for parsing only
-  extern  STP * GlobalSTP;
-  extern  STPMgr * ParserBM;
-  extern ParserInterface * parserInterface;
+  extern __thread STP * GlobalSTP;
+  extern __thread STPMgr * ParserBM;
+  extern __thread ParserInterface * parserInterface;
 
   //Some constant global vars for the Main function. Once they are
   //set, these globals will remain constants. These vars are not used
